### DISCUSSION TITLE
Não entendi trecho de codigo

### DISCUSSION THREAD

**Discussion by iago8311** [balta-io/forum#335] [Tira Dúvidas] [answered]
Na aula de "Repositórios específicos" do curso de "Acesso a Dados com .Net...." no minuto 2:49 apesar de ser explicado na aula, não entendi absolutamente nada que ele faz 
![Captura de Tela (34)](https://github.com/balta-io/forum/assets/5612445/99d37439-9808-45d4-826f-6f3b81f2b8c9)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @iago8311, como vai?

Preciso de mais informações sobre o que você não entendeu, quais linhas por exemplo, até onde conseguiu entender, de onde não conseguiu mais evoluir.

Assim posso te ajudar mais.

**Comment by iago8311** [comment]
bom dia, não entendi nada das linhas 28 a 30

**Reply by andrebaltieri** [reply]
Bom dia, @iago8311, como vai?

Vamos lá...

Na linha 28, temos a criação da variável `items` que vai armazenar o resultado da consulta no banco.
Ainda na linha 28, dizemos que a Query (Consulta) vai utilizar dois objetos, o `User` e `Role` e o resultado será armazenado em um terceiro objeto, também do tipo `User`.

Na linha 29 informamos a query (Instrução SQL) que será executada.

Na linha 30 temos a criação dos objetos que iremos utilizar. No caso chamei de user e role para ficar mais intuitivo, mas podia ser x, y, ou qualquer nome.

Na linha 32, recuperamos o usuário que receberá os dados. Provavelmente essa variável `users` foi declarada mais acima.

Caso o usuário não seja encontrado na lista anterior, utilizamos o que está sendo percorrido e então fazemos a adição do perfil a ele.

Depois disso retornamos a lista de usuários.

Veja se ficou mais claro e se tiver mais dúvidas é só responder aqui.
